### PR TITLE
Typehint to greader.php file

### PR DESCRIPTION
**Pull request by ColonelMoutarde**
Changes proposed in this pull request:

- type hint the code and cleaning

- [X] clear commit messages
- [X] code manually tested

Pass PhpStan level 7

### CODE REVIEW

File: p/api/greader.php [hunk 1]
```diff
@@ -104,17 +108,15 @@ function debugInfo(): string {
 
 final class GReaderAPI {
 
-	/** @return never */
-	private static function badRequest() {
+	private static function badRequest(): void {
```
Comment (aledeg) [inline p/api/greader.php]: I think this is not the intend. We should use `never` as a type-hint but it is not available until PHP 8.1. The same applies for other methods ending with a `die`. Maybe this applies also for methods ending with an `exit`.

Comment (ColonelMoutarde) [reply p/api/greader.php]: I understand. I can't wait for the release of the new version of debian, we will be able to type for php 7.4 at least

Comment (ColonelMoutarde) [reply p/api/greader.php]: done

**Review by aledeg — APPROVED** [review]

File: p/api/greader.php [hunk 2]
```diff
@@ -30,17 +30,21 @@
 
 if (PHP_INT_SIZE < 8) {	//32-bit
 	function hex2dec(string $hex): string {
-		if (!ctype_xdigit($hex)) return '0';
+		if (!ctype_xdigit($hex)) {
```
Comment (Alkarex) [inline p/api/greader.php]: Although I do not mind the proposed syntax, please avoid minor stylistic code changes, especially when not related to the PR.

File: p/api/greader.php [hunk 3]
```diff
@@ -425,7 +429,7 @@ private static function subscriptionEdit(array $streamNames, array $titles, stri
 					$feed = $feedDAO->searchByUrl($streamUrl);
 					$feedId = $feed == null ? -1 : $feed->id();
 				}
-				$title = isset($titles[$i]) ? $titles[$i] : '';
+				$title = $titles[$i] ?? '';
```
Comment (Alkarex) [inline p/api/greader.php:430]: Same here: I like the new syntax (which is [exactly equivalent to the former one](https://php.net/language.operators.comparison#language.operators.comparison.coalesce)), but I would rather not change it in random places, especially when not related to the PR

File: p/api/greader.php [hunk 4]
```diff
@@ -467,7 +471,7 @@ private static function subscriptionEdit(array $streamNames, array $titles, stri
 	private static function quickadd(string $url) {
 		try {
 			$url = htmlspecialchars($url, ENT_COMPAT, 'UTF-8');
-			if (substr($url, 0, 5) === 'feed/') {
+			if (0 === strpos($url, 'feed/')) {
```
Comment (Alkarex) [inline p/api/greader.php]: That is also not related to this PR, but if anything, the new [`str_starts_with()`](https://php.net/function.str-starts-with) should be used, with a polyfill in `lib_rss.php`.

File: p/api/greader.php [hunk 5]
```diff
@@ -668,9 +672,6 @@ private static function streamContents(string $path, string $include_target, int
 		header('Content-Type: application/json; charset=UTF-8');
 
 		switch ($path) {
-			case 'reading-list':
```
Comment (Alkarex) [inline p/api/greader.php:671]: Why did you remove this section? If anything, it can be moved together with default

File: p/api/greader.php (hunk 2 above)
Comment (Alkarex) [reply p/api/greader.php:33]: No need to change it back though

File: p/api/greader.php (hunk 3 above)
Comment (Alkarex) [reply p/api/greader.php:430]: No need to change it back though

File: p/api/greader.php (hunk 5 above)
Comment (ColonelMoutarde) [reply p/api/greader.php:671]: By deleting it, it is automatically directed to the default

File: p/api/greader.php (hunk 2 above)
Comment (ColonelMoutarde) [reply p/api/greader.php]: done

File: p/api/greader.php (hunk 5 above)
Comment (Alkarex) [reply p/api/greader.php:671]: Yes I know, but by removing it, you lose the information about what you are supposed to support

Comment (Frenzie) [reply p/api/greader.php:671]: Right now it's clear that it's supposed to do that. Losing that clarity is not a good thing. :-)

File: p/api/greader.php (hunk 4 above)
Comment (ColonelMoutarde) [reply p/api/greader.php]: Done

File: p/api/greader.php (hunk 5 above)
Comment (Alkarex) [reply p/api/greader.php:671]: > If anything, it can be moved together with default

Addressed in https://github.com/FreshRSS/FreshRSS/pull/5108/commits/6e330aec62107297b192e0fe65a86276a50cce88
